### PR TITLE
Use new class Lgnt

### PR DESCRIPTION
**Pull request by benoitdevos**
- Use Lgnt
- Use new addCollectionItem

logion-network/logion-internal#1107
logion-network/logion-internal#1109

### CODE REVIEW

File: src/common/TransferAmountCell.tsx [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
-    amount: Numbers.PrefixedNumber | null,
+    amount: Lgnt | null,
 }
 
 export default function TransferAmountCell(props: Props) {
+    const formatter = new LgntFormatter(2, toUnit(props.amount || Lgnt.zero()));
     if(props.amount === null) {
         return <Cell content="-" align="center" />;
-    } else if(props.amount.isNegative()) {
+    } else if(props.amount.canonical < 0n) {
```
Comment (gdethier) [inline src/common/TransferAmountCell.tsx:20]: May be good to add isNegative to Lgnt.

File: src/common/TransferAmountCell.tsx [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
         return (
             <div className="TransferAmountCell">
                 <Icon icon={{id: 'send_red'}} />
                 <span className="number" style={{color: RED}}>
-                    { props.amount.coefficient.toFixedPrecision(2) + " " + props.amount.prefix.symbol }
+                    { formatter.format(props.amount) }
```
Comment (gdethier) [inline src/common/TransferAmountCell.tsx:25]: Explicit the (default) number of decimals?

File: src/common/TransferAmountCell.tsx [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
                 </span>
             </div>
         );
-    } else if(props.amount.coefficient.isZero()) {
+    } else if(props.amount.equalTo(Lgnt.zero())) {
         return (
             <div className="TransferAmountCell" style={{justifyContent: 'flex-end'}}>
                 <span className="number">
-                    { props.amount.coefficient.toFixedPrecision(2) + " " + props.amount.prefix.symbol }
+                    { formatter.format(props.amount) }
```
Comment (gdethier) [inline src/common/TransferAmountCell.tsx:33]: Same remark as above

File: src/common/TransferAmountCell.tsx [hunk 4]
```diff
@@ -37,36 +39,36 @@ export default function TransferAmountCell(props: Props) {
...
     }
 }
 
-export function transferBalance(address: string, transaction: Transaction): Numbers.PrefixedNumber {
-    const amount = toPrefixedLgnt(transaction.transferValue);
+export function transferBalance(address: string, transaction: Transaction): Lgnt {
+    const amount = Lgnt.fromCanonical(BigInt(transaction.transferValue));
     if(transaction.from === address) {
-        return amount.negate();
+        return amount.multiply(-1n); // TODO use negate() when available
```
Comment (gdethier) [inline src/common/TransferAmountCell.tsx:52]: Yes indeed!

**Review by gdethier — APPROVED** [review]
:+1:

File: src/common/TransferAmountCell.tsx (hunk 1 above)
Comment (benoitdevos) [reply src/common/TransferAmountCell.tsx:20]: Done.

File: src/common/TransferAmountCell.tsx (hunk 4 above)
Comment (benoitdevos) [reply src/common/TransferAmountCell.tsx:52]: Done.

File: src/common/TransferAmountCell.tsx (hunk 2 above)
Comment (benoitdevos) [reply src/common/TransferAmountCell.tsx:25]: Not sure to understand what you mean - the number of decimals is given as a parameter of the formatter, 8 lines above this one. As it does not look critical, I will temporarily leave it unchanged.

Comment (gdethier) [reply src/common/TransferAmountCell.tsx:25]: Indeed, my bad!

**Pull request merged**